papaya-web/src/main/java/pl/edu/agh/papaya/rest/sprints/controller/SprintsController.java

```diff
@@ -0,0 +1,49 @@
+package pl.edu.agh.papaya.rest.sprints.controller;
+
+import com.google.common.collect.Lists;
+import java.time.LocalDateTime;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+import pl.edu.agh.papaya.api.model.SprintStateDto;
+import pl.edu.agh.papaya.api.model.SprintWithStateDto;
+import pl.edu.agh.papaya.api.service.SprintsApi;
+import pl.edu.agh.papaya.mappers.SprintMapper;
+import pl.edu.agh.papaya.mappers.SprintStateMapper;
+import pl.edu.agh.papaya.model.SprintState;
+import pl.edu.agh.papaya.service.sprint.SprintService;
+
+@RestController
+public class SprintsController implements SprintsApi {
+
+    @Autowired
+    private SprintService sprintService;
+
+    @Autowired
+    private SprintStateMapper sprintStateMapper;
+
+    @Autowired
+    private SprintMapper sprintMapper;
+
+    @Override
+    public ResponseEntity<List<SprintWithStateDto>> getSprints(List<SprintStateDto> sprintStateDtos) {
+        final LocalDateTime currentTime = LocalDateTime.now();
+
+        List<SprintState> sprintStates = sprintStateMapper.mapFromApi(
+                Optional.ofNullable(sprintStateDtos).orElseGet(() -> Lists.newArrayList(SprintStateDto.values())));
+
+        List<SprintWithStateDto> sprints = sprintService.findByStates(sprintStates, currentTime)
+                .stream()
+                .sorted(Comparator.comparing(sprint -> sprint.getEnrollmentPeriod().getStart()))
+                .map(sprint -> new SprintWithStateDto()
+                        .sprint(sprintMapper.mapToApi(sprint))
+                        .sprintState(sprintStateMapper.mapToApi(sprint.getSprintState(currentTime))))
```

Comment:
może jakiś dedykowany konwerter?

Reply:
Poczytam internet

Reply:
A po co czytać internet xd 
Chodziło mi o te mappery (źle się wyraziłem), żeby zamiast tworzenia obiektu w streamie, było odwołanie do mapToApi. Wtedy można się też pozbyć dwóch mapperów z tej klasy i zastąpić go jednym. 

Trzeba jeszcze przemyśleć, czy podawanie w mapperze tego currentTime spełnia nasze kryteria _fajności_

Reply:
Aaaa, w sensie `SprintWithStateSto` przestaje istnieć

Reply:
Zaraz pushnę

Reply:
Jak nie podasz tego czasu, to między query na bazie i `getSprintState` może się zmienić stan, a zakładam, że interesuje nas Jakiś Moment

Reply:
Można by to jakoś odwrócić i nie pytać `SprintService` o listę stanów, tylko o każdy po kolei i już wtedy im ustawiać, ale po coooo

Reply:
wiem, wiem
chodziło mi o to, że zawsze trzeba to wtedy podawać i nie wiem, czy takie rozwiązanie będzie okej (w sensie ten cały konwerter)

Reply:
Jest też wariant bez, który sobie bierze `LocalDateTime.now()`

Reply:
wtedy trudno będzie to przetestować

Reply:
No są dwa warianty jak w przypadku wszystkich tych metod, które robią cokolwiek ze `SprintState`/`SprintStateDto`

Reply:
a okej, przeoczyłem